pyrefinebio/high_level_functions.py

```diff
@@ -120,12 +121,26 @@ def download_dataset(
 
     dataset.process()
 
+    start_time = datetime.now()
     while not dataset.check():
+        if not dataset.is_processing and not dataset.is_processed:
+            print(
+                "Dataset failed to process. The system may be experiencing issues."
+                " Please try again later."
+            )
+            return None
+        elif datetime.now() - start_time > timedelta(minutes=15):
+            print(
+                "Dataset not processed after 15 minutes. The system may be experiencing issues."
```

Comment:
How long do most datasets take to smash? If someone provides a somewhat large dataset that takes more than 15 minutes to smash, so they would get a message that "The system may be experiencing issues. Please try again later.", which might cause them to think that the system is broken so they should resubmit the job.

Reply:
Hmm, this is actually a really good question. It looks like some of the smasher jobs have taken hours. Now I'm not sure what I want to do about this....